#
# Submitted-By: Marc Kleine-Budde, 2006-04-21
# Committed-By: Marc Kleine-Budde
#
# Error:
#
# libtool sux, cannont crosscompile if -l<lib> is given, add -Wl, prefixes
#
Index: sablevm-1.13/configure.ac
===================================================================
--- sablevm-1.13.orig/configure.ac
+++ sablevm-1.13/configure.ac
@@ -314,7 +314,7 @@ AC_LIBTOOL_DLOPEN
 AC_PROG_LIBTOOL
 
 dnl Checks for libraries.
-LIBSVMADD="-lm -lltdl -lz"
+LIBSVMADD="-Wl,-lm -Wl,-lltdl -Wl,-lz"
 AC_CHECK_LIB(ltdl,lt_dlinit,echo -n,echo ***ERROR: libltdl is missing; exit 1)
 AC_CHECK_LIB(m,fmod,echo -n,echo ***ERROR: libm is missing; exit 1)
 AC_CHECK_LIB(z,inflate,echo -n,echo ***ERROR: zlib/libz is missing; exit 1)
@@ -323,7 +323,7 @@ dnl Check if -lrt is needed
 case $host in
 	*-sun-solaris*)
 	    dnl we should call AC_CHECK_LIB here, but what symbol should we look up?
-	    LIBSVMADD="$LIBSVMADD -lrt"
+	    LIBSVMADD="$LIBSVMADD -Wl,-lrt"
 	    ;;
 	*)  ;;
 esac
@@ -335,7 +335,7 @@ case $host in
 	    ;;
 	*)
 	    AC_CHECK_LIB(pthread,pthread_self,echo -n,echo ***ERROR: libpthread is missing; exit 1)
-	    LIBSVMADD="$LIBSVMADD -lpthread"
+	    LIBSVMADD="$LIBSVMADD -Wl,-lpthread"
 	    ;;
 esac
 
@@ -348,8 +348,8 @@ if test "$with_internal_libffi" = "auto"
 fi
 case "$with_internal_libffi" in
 	no)	AC_CHECK_LIB(ffi,ffi_prep_cif,echo -n,echo ***ERROR: libffi is missing; exit 1)
-		LIBSVMADD="$LIBSVMADD -lffi";;
-	no2)	LIBSVMADD="$LIBSVMADD -lffi";; dnl we already checked the lib above
+		LIBSVMADD="$LIBSVMADD -Wl,-lffi";;
+	no2)	LIBSVMADD="$LIBSVMADD -Wl,-lffi";; dnl we already checked the lib above
 	yes)	ac_configure_args="$ac_configure_args --enable-convenience-only-libffi"
 		LIBSVMADD="$LIBSVMADD ../libffi/libffi_convenience.la"
 		LIBSVMINCLUDE="-I../libffi -I../libffi/include"
@@ -372,8 +372,8 @@ if test "$with_internal_libpopt" = "auto
 fi
 case "$with_internal_libpopt" in
 	no)	AC_CHECK_LIB(popt,poptGetContext,echo -n,echo ***ERROR: libpopt is missing; exit 1)
-		LIBSVMADD="$LIBSVMADD -lpopt";;
-	no2)	LIBSVMADD="$LIBSVMADD -lpopt";;dnl we already checked the lib above
+		LIBSVMADD="$LIBSVMADD -Wl,-lpopt";;
+	no2)	LIBSVMADD="$LIBSVMADD -Wl,-lpopt";;dnl we already checked the lib above
 	yes)	ac_configure_args="$ac_configure_args --enable-convenience-only-libpopt"
 		SVMADD="$LIBSVMADD ../libpopt/libpopt_convenience.la"
 		SVMINCLUDE='-I../libpopt -I$(srcdir)/../libpopt'
